tests/conftest.py

```diff
@@ -124,57 +125,100 @@ def clean_serveradmin(filters):
     Query(filters).delete().commit()
 
 
-def clean_aws(vm_hostname):
-    def _get_instance_status():
-        response = ec2.describe_instances(
-            Filters=[
-                {
-                    'Name': 'instance-state-code',
-                    'Values': [
-                        str(AWS_RETURN_CODES['pending']),
-                        str(AWS_RETURN_CODES['running']),
-                        str(AWS_RETURN_CODES['shutting-down']),
-                        str(AWS_RETURN_CODES['terminated']),
-                        str(AWS_RETURN_CODES['stopping']),
-                        str(AWS_RETURN_CODES['stopped']),
-                    ]
-                },
-            ],
-            InstanceIds=[obj['aws_instance_id']],
-            DryRun=False)['Reservations'][0]['Instances'][0]['State']['Code']
-
-        return int(response)
+def _get_aws_instance_status(instance_id: str) -> int:
+    ec2 = boto3.client('ec2')
+    response = ec2.describe_instances(
+        Filters=[
+            {
+                'Name': 'instance-state-code',
+                'Values': [
+                    str(AWS_RETURN_CODES['pending']),
+                    str(AWS_RETURN_CODES['running']),
+                    str(AWS_RETURN_CODES['shutting-down']),
+                    str(AWS_RETURN_CODES['terminated']),
+                    str(AWS_RETURN_CODES['stopping']),
+                    str(AWS_RETURN_CODES['stopped']),
+                ]
+            },
+        ],
+        InstanceIds=[instance_id],
+        DryRun=False)['Reservations'][0]['Instances'][0]['State']['Code']
+
+    return int(response)
 
+
+def clean_aws(vm_hostname):
     try:
         obj = Query({'hostname': vm_hostname}, ['aws_instance_id']).get()
     except DatasetError:  # No object to clean up
         return
 
+    timeout = 120
+    ec2 = boto3.client('ec2')
```

Comment:
why exactly was this moved above the if? those variables are not required for the if right?

Reply:
refactored the cleanup